### PR TITLE
Improve Documentation VII

### PR DESCRIPTION
**Pull request by zm711**
So in this documentation improvement I did:

1) My usual search for typo fixes/minor clarifications
2) I rearranged the sorters docs to move the internal sorters above the legacy sorters because as a reader once I reach legacy I would likely stop reading the docs rather than read on to see SC2 and Tridesclous2, so it is better to be proud of the internal products and move them above legacy. I also added install instruction for Tridesclous2 so for people that want to test it out they can
3) I deleted the legacy section of widgets documentation and instead moved/made sure that every widget currently exposed in the `widgets_list.py` is connected to in the docs. My question for that is: I added `matplotlib` backend, but do all widgets now all have at least the `ipy` backend as well
4) I actually went through the examples for the modules gallery to fix typos and try to add clarifications and keyword arguments

### CODE REVIEW

**Comment by alejoe91** [comment]
Thank you so much Zach!!! This is great :)

No not all widgets have a `ipy` backend yet. For example:
- `plot_all_amplitudes_distributions`
- `plot_crosscorrelograms`
- `plot_agreement_scores`

The available backends are correctly listed in the API references: https://spikeinterface.readthedocs.io/en/latest/api.html#spikeinterface.widgets.plot_all_amplitudes_distributions

**Comment by zm711** [comment]
Of course :)

> The available backends are correctly listed in the API references: 
https://spikeinterface.readthedocs.io/en/latest/api.html#spikeinterface.widgets.plot_all_amplitudes_distributions
Cool, I'll update the backend part of the docs soon.

I wanted to update this because I constantly go to that section of the docs to remember the exact name of the plotting function I want (for example the name change from _multicomp_ to multicomparison etc).

**Comment by alejoe91** [comment]
> Of course :)
> 
> > The available backends are correctly listed in the API references:
> > https://spikeinterface.readthedocs.io/en/latest/api.html#spikeinterface.widgets.plot_all_amplitudes_distributions
> 
> Cool, I'll update the backend part of the docs soon.
> 
> I wanted to update this because I constantly go to that section of the docs to remember the exact name of the plotting function I want (for example the name change from _multicomp_ to multicomparison etc).

@zm711 since you're at it, could you remove the Legacy part of the widgets API (https://spikeinterface--2229.org.readthedocs.build/en/2229/api.html#legacy-widgets) and check if all plot_functions are listed there?

[List of plot functions](https://github.com/SpikeInterface/spikeinterface/blob/main/src/spikeinterface/widgets/widget_list.py#L107-L142)

[API entries](https://github.com/SpikeInterface/spikeinterface/blob/main/doc/api.rst)

**Comment by zm711** [comment]
Yep. I can do that as well. It's a holiday coming up here so it might not happen this week, but I'll put it on my todo list.

**Comment by alejoe91** [comment]
No rush! I can also try to add them if you're ok with me pushing here :)

**Comment by zm711** [comment]
> No rush! I can also try to add them if you're ok with me pushing here :)

Of course. Always.

**Review by alejoe91 — APPROVED** [review]

**Comment by alejoe91** [comment]
Done and also fixed some other sphinx warnings and missing refs :)

File: doc/modules/sorters.rst [hunk 1]
```diff
@@ -12,9 +12,9 @@ On the other hand SpikeInterface directly implements some internal sorters (**sp
 that do not depend on external tools, but depend on the :py:mod:`spikeinterface.sortingcomponents`
 module. **Note that internal sorters are currently experimental and under development**.
 
-A drawback of using external sorters is the installation of these tools. Sometimes they need MATLAB,
-specific versions of CUDA, specific gcc versions vary or even worse outdated versions of
-Python/NumPy. In that case, SpikeInterface offer the mechanism of running external sorters inside a
+A drawback of using external sorters is the separate installation of these tools. Sometimes they need MATLAB,
+specific versions of CUDA, specific gcc versions or outdated versions of
+Python/NumPy. In this case, SpikeInterface offer the mechanism of running external sorters inside a
```
Comment (alejoe91) [inline doc/modules/sorters.rst:17]:
```suggestion
Python/NumPy. In this case, SpikeInterface offers the mechanism of running external sorters inside a
```

File: doc/modules/widgets.rst [hunk 2]
```diff
@@ -70,15 +71,15 @@ To install it, run:
 
 .. code-block:: bash
 
-    pip install sortingview figurl-jupyter
+    pip install sortingview
```
Comment (alejoe91) [inline doc/modules/widgets.rst:74]: nice catch!

File: doc/viewers.rst [hunk 3]
```diff
@@ -16,15 +16,15 @@ spikeinterface.widgets
 
 The easiest way to visualize :code:`spikeinterface` objects is to use the :code:`widgets` module for plotting.
 You can find an extensive description in the module documentation :ref:`modulewidgets`
-and many examples in this tutorial :ref:`sphx_glr_modules_gallery_widgets`.
```
Comment (alejoe91) [inline doc/viewers.rst:19]: For context, this was generating a missing reference and I couldn't find a way to ref a dynamically generated sphinx-gallery module, but just individual files.

Comment (zm711) [reply doc/viewers.rst:19]: A lot of the sphinx gallery stuff is annoying like that, but based on my reading they don't really have plans to try to patch things like that. Such is life I suppose.

File: doc/modules/core.rst [hunk 4]
```diff
@@ -585,12 +610,12 @@ In this example, we create a recording and a sorting object from numpy objects:
...
-:py:class:`~spikeinterface.core.SharedMemorySorting` easily like this
+:py:class:`~spikeinterface.core.SharedMemorySorting` easily like this:
 
 .. code-block:: python
 
     # turn any sortinto into NumpySorting
-    soring_np = sorting.to_numpy_sorting()
+    sorting_np = sorting.to_numpy_sorting()
 
     # or to SharedMemorySorting for parrallel computing
```
Comment (zm711) [inline doc/modules/core.rst:620]:
```suggestion
    # or to SharedMemorySorting for parallel computing
```

**Review by zm711 — COMMENTED** [review]
Couple more typos while we are at it.

**Review by alejoe91 — APPROVED** [review]

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**